### ISSUE TITLE
missing FROM-clause entry for table "subquery"

### ISSUE THREAD

**Issue by RyVeata** [solspace/craft3-calendar#238]
Following an upgrade of a website from Craft 3 to Craft 4, the Solspace Calendar plugin was successfully updated to version 4.0.9 but on accessing the calendar events area a server error occurs returning the response `missing FROM-clause entry for table "subquery"` (full print out below).  This seems to occur for all occurrences of querying on events for my build.

**Steps to reproduce**
(apologies for the lack of clarity here)

1. Have a prexisting install of Craft with Solspace Calendar (originally version 3) and pre-existing events.  In my scenario the events are split across a number of different calendars.
2. Upgrade the version to 4.0.9
3. Go to the Events section of the Calendar control panel, no events are listed and a server error notice occurs

**Expected behavior**
The control panel Events section should load a list of all available events for the selected calendar.  Events should be able to be displayed on the front end.

**Craft & Plugin Info (please complete the following information):**

- Craft Version: 4.5.6
- Calendar Version: 4.0.9
- Calendar Edition: Pro (trial)
- Fresh Install or Upgrade: Upgrade
- Yii version 2.0.48.1

**Additional context**

From my own investigation I have established that the subqueries in the EventsService.php once built into SQL by Yii are defined with names encased in double quotes but when queried against are not and so it can't seem to establish what subquery is.  By wrapping the references to subquery in the outer query with quotes this issue is resolved.

For example line 213 on EventsService.php goes from:
`->innerJoin(ElementRecord::tableName().' elements', 'elements.[[id]] = subQuery.[[elementsId]]')`
to
`->innerJoin(ElementRecord::tableName().' elements', 'elements.[[id]] = "subQuery".[[elementsId]]')`

If applied to all join lines referencing subQuery the issue is resolved and the plugin functions correctly.

Error returned reads as follows:
```
2023-09-27 11:30:57 [web.ERROR] [yii\db\Exception] PDOException: SQLSTATE[42P01]: Undefined table: 7 ERROR:  missing FROM-clause entry for table "subquery"
LINE 9: ...NER JOIN "elements" "elements" ON elements."id" = subQuery."...
                                                             ^ in /usr/share/nginx/vendor/yiisoft/yii2/db/Command.php:1302
Stack trace:
#0 /usr/share/nginx/vendor/yiisoft/yii2/db/Command.php(1302): PDOStatement->execute()
#1 /usr/share/nginx/vendor/yiisoft/yii2/db/Command.php(1168): yii\db\Command->internalExecute()
#2 /usr/share/nginx/vendor/yiisoft/yii2/db/Command.php(410): yii\db\Command->queryInternal()
#3 /usr/share/nginx/vendor/yiisoft/yii2/db/Query.php(249): yii\db\Command->queryAll()
#4 /usr/share/nginx/vendor/craftcms/cms/src/db/Query.php(247): yii\db\Query->all()
#5 /usr/share/nginx/vendor/solspace/craft-calendar/packages/plugin/src/Services/EventsService.php(161): craft\db\Query->all()
#6 /usr/share/nginx/vendor/solspace/craft-calendar/packages/plugin/src/Elements/Db/EventQuery.php(936): Solspace\Calendar\Services\EventsService->getSingleEventMetadata()
#7 /usr/share/nginx/vendor/solspace/craft-calendar/packages/plugin/src/Elements/Db/EventQuery.php(517): Solspace\Calendar\Elements\Db\EventQuery->cacheSingleEvents()
#8 /usr/share/nginx/vendor/craftcms/cms/src/base/Element.php(1243): Solspace\Calendar\Elements\Db\EventQuery->all()
#9 /usr/share/nginx/vendor/craftcms/cms/src/base/Element.php(1193): craft\base\Element::indexElements()
#10 /usr/share/nginx/vendor/craftcms/cms/src/controllers/ElementIndexesController.php(701): craft\base\Element::indexHtml()
#11 /usr/share/nginx/vendor/craftcms/cms/src/controllers/ElementIndexesController.php(165): craft\controllers\ElementIndexesController->elementResponseData()
#12 [internal function]: craft\controllers\ElementIndexesController->actionGetElements()
#13 /usr/share/nginx/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array()
#14 /usr/share/nginx/vendor/yiisoft/yii2/base/Controller.php(178): yii\base\InlineAction->runWithParams()
#15 /usr/share/nginx/vendor/yiisoft/yii2/base/Module.php(552): yii\base\Controller->runAction()
#16 /usr/share/nginx/vendor/craftcms/cms/src/web/Application.php(304): yii\base\Module->runAction()
#17 /usr/share/nginx/vendor/craftcms/cms/src/web/Application.php(607): craft\web\Application->runAction()
#18 /usr/share/nginx/vendor/craftcms/cms/src/web/Application.php(283): craft\web\Application->_processActionRequest()
#19 /usr/share/nginx/vendor/yiisoft/yii2/base/Application.php(384): craft\web\Application->handleRequest()
#20 /usr/share/nginx/web/index.php(12): yii\base\Application->run()
#21 {main}

Next yii\db\Exception: SQLSTATE[42P01]: Undefined table: 7 ERROR:  missing FROM-clause entry for table "subquery"
LINE 9: ...NER JOIN "elements" "elements" ON elements."id" = subQuery."...
                                                             ^
The SQL being executed was: SELECT events."id", events."startDate", elements_sites."siteId", content."title"
FROM (SELECT elements."id" AS "elementsId", elements_sites."id" AS "elementsSitesId", content."id" AS "contentId"
FROM "elements" "elements"
INNER JOIN "calendar_events" "events" ON events."id" = elements."id"
INNER JOIN "calendar_calendars" "calendars" ON calendars."id" = events."calendarId"
INNER JOIN "elements_sites" "elements_sites" ON elements_sites."elementId" = elements."id"
INNER JOIN "content" "content" ON (content."elementId" = elements."id") AND ("content"."siteId" = elements_sites."siteId")
WHERE (events."freq" IS NULL) AND (elements."dateDeleted" IS NULL) AND (events."id" IN (36208, 30492, 30239, 6889, 31888, 21400, 4668, 23076, 11438, 11677, 10863, 10865, 6758, 21431, 33390, 23025, 6983, 11576, 7017, 12569, 27580, 21367, 5441, 9971, 33393, 17342, 6891, 6788, 762, 30560, 12360, 6886, 11434, 6906, 34961, 34748, 36647, 9974, 7931, 34724, 12574, 21254, 37707, 37942, 34194, 34034, 34026, 34196, 37917, 6845, 24273, 24279, 5761, 8202, 10824, 33914, 11574, 33754, 10732, 7006, 7009, 22397, 5466, 9976, 31884, 34750)) AND (elements_sites."siteId"=3)) "subQuery"
INNER JOIN "elements" "elements" ON elements."id" = subQuery."elementsId"
INNER JOIN "elements_sites" "elements_sites" ON elements_sites."id" = subQuery."elementsSitesId"
INNER JOIN "calendar_events" "events" ON events."id" = subQuery."elementsId"
INNER JOIN "calendar_calendars" "calendars" ON calendars."id" = events."calendarId"
INNER JOIN "content" "content" ON content."id" = subQuery."contentId" in /usr/share/nginx/vendor/yiisoft/yii2/db/Schema.php:676
Stack trace:
#0 /usr/share/nginx/vendor/yiisoft/yii2/db/Command.php(1307): yii\db\Schema->convertException()
#1 /usr/share/nginx/vendor/yiisoft/yii2/db/Command.php(1168): yii\db\Command->internalExecute()
#2 /usr/share/nginx/vendor/yiisoft/yii2/db/Command.php(410): yii\db\Command->queryInternal()
#3 /usr/share/nginx/vendor/yiisoft/yii2/db/Query.php(249): yii\db\Command->queryAll()
#4 /usr/share/nginx/vendor/craftcms/cms/src/db/Query.php(247): yii\db\Query->all()
#5 /usr/share/nginx/vendor/solspace/craft-calendar/packages/plugin/src/Services/EventsService.php(161): craft\db\Query->all()
#6 /usr/share/nginx/vendor/solspace/craft-calendar/packages/plugin/src/Elements/Db/EventQuery.php(936): Solspace\Calendar\Services\EventsService->getSingleEventMetadata()
#7 /usr/share/nginx/vendor/solspace/craft-calendar/packages/plugin/src/Elements/Db/EventQuery.php(517): Solspace\Calendar\Elements\Db\EventQuery->cacheSingleEvents()
#8 /usr/share/nginx/vendor/craftcms/cms/src/base/Element.php(1243): Solspace\Calendar\Elements\Db\EventQuery->all()
#9 /usr/share/nginx/vendor/craftcms/cms/src/base/Element.php(1193): craft\base\Element::indexElements()
#10 /usr/share/nginx/vendor/craftcms/cms/src/controllers/ElementIndexesController.php(701): craft\base\Element::indexHtml()
#11 /usr/share/nginx/vendor/craftcms/cms/src/controllers/ElementIndexesController.php(165): craft\controllers\ElementIndexesController->elementResponseData()
#12 [internal function]: craft\controllers\ElementIndexesController->actionGetElements()
#13 /usr/share/nginx/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array()
#14 /usr/share/nginx/vendor/yiisoft/yii2/base/Controller.php(178): yii\base\InlineAction->runWithParams()
#15 /usr/share/nginx/vendor/yiisoft/yii2/base/Module.php(552): yii\base\Controller->runAction()
#16 /usr/share/nginx/vendor/craftcms/cms/src/web/Application.php(304): yii\base\Module->runAction()
#17 /usr/share/nginx/vendor/craftcms/cms/src/web/Application.php(607): craft\web\Application->runAction()
#18 /usr/share/nginx/vendor/craftcms/cms/src/web/Application.php(283): craft\web\Application->_processActionRequest()
#19 /usr/share/nginx/vendor/yiisoft/yii2/base/Application.php(384): craft\web\Application->handleRequest()
#20 /usr/share/nginx/web/index.php(12): yii\base\Application->run()
#21 {main}
```

**Comment by kjmartens** [comment]
Sorry for the delay and the trouble you're experiencing, @RyVeata.

I haven't seen an issue like this reported before, so I'm not sure why it's specifically affecting you. At any rate, I will have a developer check into this and make any changes if necessary. 🙂

**Comment by kjmartens** [comment]
This is now resolved in Calendar 4.0.11+. 🙂
